### PR TITLE
[Ming-Omni] talker duration guard fix

### PR DESCRIPTION
**Pull request by AkazaAkane**
## Motivation

The Ming-Omni talker duration guard previously checked accumulated audio length by repeatedly calling `torch.cat(all_wavs, dim=-1).shape[...]` inside the streaming loop.

That is unnecessarily expensive: every duration check materializes a new concatenated waveform tensor and recopies all chunks generated so far. For multi-chunk generations, this makes the guard O(n²) in accumulated audio size even though the guard only needs the current sample count.

This PR replaces those repeated concat-based length checks with a lightweight running sample counter.

## Modifications

* Add `total_samples` inside `MingOmniTalker._process_segment`.
* Increment `total_samples` whenever a generated `tts_speech` chunk is appended/yielded.
* Replace per-chunk `torch.cat(all_wavs, dim=-1).shape[...]` duration checks with `total_samples / sample_rate`.
* Preserve the existing guard placement and duration-limit semantics.
* Keep chunk yielding behavior unchanged.
* Add `_run_process_segment_with_chunk_lengths` test helper for deterministic duration-guard tests.
* Add unit tests for:

  * single chunk below the guard
  * multi-chunk generation below the guard
  * stopping after the chunk that crosses the duration guard
  * strict two-second boundary behavior
  * empty generation
  * ensuring the duration guard does not call `torch.cat` per chunk

## Related Issues

Refs #999.

## Accuracy Test

all tests passed

## Benchmark & Profiling

No full model benchmark was run for this PR.

Expected performance impact is localized to long / multi-chunk Ming talker generations:

* before: duration guard repeatedly concatenated all prior waveform chunks
* after: duration guard uses an O(1) running sample counter per chunk
* final audio assembly behavior is unchanged

The regression test monkeypatches `torch.cat` to raise and confirms the duration guard no longer depends on per-chunk concatenation.

## Checklist

* [ ] Format your code according with pre-commit.
* [x] Add unit tests.
* [ ] Update documentation / docstrings / example tutorials as needed.
* [ ] Provide throughput / latency benchmark results and accuracy evaluation results as needed.
* [ ] For reviewers: If you haven't made any contributions to this PR and are only assisting with merging the main branch, please remove yourself as a co-author when merging the PR.

## CI

CI runs on self-hosted GPU runners and requires a maintainer to add the `run-ci` label. Once labeled, every subsequent push re-triggers CI as long as the label remains. Use `/tag-and-rerun-ci higgs` or `/tag-and-rerun-ci moss` to select a TTS CI model. Draft PRs are skipped even if labeled.

### CODE REVIEW

File: tests/unit_test/ming_omni/test_talker_voice_validation.py [hunk 1]
```diff
@@ -288,6 +288,42 @@ def _fake_detokenizer(sample_rate=44100, vae_patch_size=4, hop_size=480):
...
+            talker,
+            text,
+            prompt=None,
+            instruction=None,
+            spk_emb=None,
+            audio_detokenizer=detok,
+            prompt_text=None,
+            prompt_wav_lat=None,
+            prompt_wav_emb=None,
+            stream=True,
```
Comment (edwingao28) [inline tests/unit_test/ming_omni/test_talker_voice_validation.py]: Could we add a stream=False case here? The duration guard has separate streaming and aggregate paths, but these tests only cover the streaming one.

File: sglang_omni/models/ming_omni/talker/modeling_ming_omni_talker.py [hunk 2]
```diff
@@ -1273,6 +1273,7 @@ def _process_segment(
 
             use_stream = stream
             all_wavs: list = []
+            total_samples = 0
```
Comment (edwingao28) [inline sglang_omni/models/ming_omni/talker/modeling_ming_omni_talker.py:1275]: all_wavs is no longer read now that the guard uses total_samples. Could we remove it and the two appends? Keeping it retains every emitted chunk until _process_segment exits.

Comment (AkazaAkane) [reply sglang_omni/models/ming_omni/talker/modeling_ming_omni_talker.py:1275]: Removed unused all_wavs retention and appends

File: tests/unit_test/ming_omni/test_talker_voice_validation.py (hunk 1 above)
Comment (AkazaAkane) [reply tests/unit_test/ming_omni/test_talker_voice_validation.py]: Added a stream=False duration-guard test case by parameterizing the helper

**Review by edwingao28 — APPROVED** [review]

**Comment by edwingao28** [comment]
lgtm and merge this pr as this is minor fix for ming only and passed all ci
